src/Nethermind/Nethermind.TxPool/Collections/SortedPool.cs

```diff
@@ -356,16 +377,19 @@ protected virtual bool Remove(TKey key, TValue value)
         {
             if (_cacheMap.Remove(key))
             {
+                UpdateIsFull();
                 _snapshot = null;
                 return true;
             }
 
             return false;
         }
 
+        public bool IsFull() => _isFull;
 
-        [MethodImpl(MethodImplOptions.Synchronized)]
-        public bool IsFull() => _cacheMap.Count >= _capacity;
```

Comment:
Why was this synchronized in first place? Is it possible to have multi-threaded access? If yes, then more synchronization is required. If not, this is fine.

Reply:
It is possible, but it is fine to have a kind of `eventual consistency` as it is used for early filters, while later actual modification will be under lock. It is fine to be slightly off in the filter or slightly go over desired transaction pool capacity.

Reply:
> Is it possible to have multi-threaded access?

Yes

> If yes, then more synchronization is required. 

`IsFull` is now a bool set under lock. It just needs a taste, doesn't need to be accurate